docs/src/proposals.md

```diff
@@ -1 +1,120 @@
+Get Vote Counts: Retrieves the vote counts (yay and nay) for a specific proposal.
+Get Proposal Status: This checks the current status of a proposal (active, passed, or failed).
+Get Live Proposals: This returns all active proposals in an array of IDs.
+Check User Vote: Checks if a specific user has voted on a specific proposal and their voting status.
+
+# Integration
+
+The Proposals component can be integrated into your protocol without any dependencies on other Konoha components. This means it can be used standalone within the Konoha ecosystem or potentially integrated into other systems that follow similar patterns.
+
+## Key Functions
+
+##### Submitting a Proposal
+
+fn submit_proposal(
+ref self: TContractState, payload: felt252, to_upgrade: ContractType
+) -> felt252;
+
+###### Parameters:
+
+payload: The data related to the proposal.
+to_upgrade: The type of contract upgrade proposed.
+Returns: A unique identifier for the submitted proposal.
+
+##### Voting on a Proposal
+
+fn vote(
+ref self: TContractState,
+prop_id: felt252,
+opinion: felt252);
+
+###### Parameters:
+
+prop_id: The identifier of the proposal.
+opinion: The user's vote (1 for yay, 2 for nay).
+
+##### Getting Proposal Details
+
+fn get_proposal_details(self: @TContractState,
+prop_id: felt252) -> PropDetails;
+
+###### Parameters:
+
+prop_id: The identifier of the proposal.
+Returns: The details of the specified proposal.
+
+##### Getting Vote Counts
+
+fn get_vote_counts(self: @TContractState,
+prop_id: felt252) -> (u128, u128);
+
+###### Parameters:
+
+prop_id: The identifier of the proposal.
+Returns: A tuple containing the yay and nay vote counts.
+
+##### Getting Proposal Status
+
+fn get_proposal_status(self: @TContractState,
+prop_id: felt252) -> felt252;
+
+###### Parameters:
+
+prop_id: The identifier of the proposal.
+Returns: The status of the proposal (e.g., active, passed, failed).
+
+##### Getting Live Proposals
+
+fn get_live_proposals(self: @TContractState) -> Array<felt252>;
+Returns: An array of identifiers for active proposals.
+
+##### Checking User Vote
+
+fn get_user_voted(
+self: @TContractState,
+user_address: ContractAddress,
+prop_id: felt252
+) -> VoteStatus;
+
+## Parameters:
+
+user_address: The address of the user.
+prop_id: The identifier of the proposal.
+Returns: The voting status of the user for the specified proposal.
+
+## Customization
+
+While the Proposals component is robust in its default form, you may want to customize it to better fit your protocol’s needs. Here are some ways you can customize it:
+
+Custom Voting Logic: This modifies the vote function to implement custom voting logic or add additional checks.
+
+Proposal Validation: This extends the submit_proposal function to include additional validation or metadata for proposals.
+
+Extended Events: Add or modify events to capture more details about proposals and votes.
```

Comment:
The component should not be altered – it should only be mixed into a contract. Only customizations possible currently are to integrate it with Upgrades for custom proposals and in the future, to alter the voting – though this is planned. Perhaps mention that now is a good time to reach out to us regarding any things users would like to see customized